chainer/distribution.py

```diff
@@ -0,0 +1,420 @@
+
+        >>> import chainer
+        >>> import chainer.distributions as D
+        >>> import numpy as np
+        >>> d = 2
+        >>> shape = (4, 3)
+        >>> loc = np.random.normal(
+        ...     size=shape + (d,)).astype(np.float32)
+        >>> cov = np.random.normal(size=shape + (d, d)).astype(np.float32)
+        >>> cov = np.matmul(cov, np.rollaxis(cov, -1, -2))
+        >>> l = np.linalg.cholesky(cov)
+        >>> dist = D.MultivariateNormal(loc, l)
+        >>> dist.event_shape
+        (2,)
+        >>> dist.batch_shape
+        (4, 3)
+        >>> sample = dist.sample(sample_shape=(6, 5))
+        >>> sample.shape
+        (6, 5, 4, 3, 2)
+
+    Every function about probability takes realization value whose shape is
+    `(sample_shape, batch_shape, event_shape)` and returns an evaluated value
+    whose shape is `(sample_shape, batch_shape)`.
+
+    """
+
+    def _copy_to(self, target):
+        target.__dict__ = copy.copy(self.__dict__)
+        return target
+
+    @property
+    def batch_shape(self):
+        """Returns the shape of a batch.
+
+        Returns:
+            tuple: The shape of a sample that is not identical and indipendent.
+
+        """
+        raise NotImplementedError
+
+    def cdf(self, x):
+        """Evaluates the cumulative distribution function at a given points.
+
+        Args:
+            x(:class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+            :class:`cupy.ndarray`): Data points in the domain of the
+            distribution
+
+        Returns:
+            ~chainer.Variable: Cumulative distribution function value evaluated
+            at `x`.
+
+        """
+        raise NotImplementedError
+
+    @property
+    def covariance(self):
+        """Returns the covariance of the distribution.
+
+        Returns:
+            ~chainer.Variable: The covariance of the distribution.
+        """
+        raise NotImplementedError
+
+    @property
+    def entropy(self):
+        """Returns the entropy of the distribution.
+
+        Returns:
+            ~chainer.Variable: The entropy of the distribution.
+
+        """
+        raise NotImplementedError
+
+    @property
+    def event_shape(self):
+        """Returns the shape of an event.
+
+        Returns:
+            tuple: The shape of a sample that is not identical and indipendent.
+
+        """
+        raise NotImplementedError
+
+    def icdf(self, x):
+        """Evaluates the inverse cumulative distribution function at a given points.
+
+        Args:
+            x(:class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+            :class:`cupy.ndarray`): Data points in the domain of the
+            distribution
+
+        Returns:
+            ~chainer.Variable: Inverse cumulative distribution function value
+            evaluated at `x`.
+
+        """
+        raise NotImplementedError
+
+    def log_cdf(self, x):
+        """Evaluates the log of cumulative distribution function at a given points.
```

Comment:
"at a given points" -> "at the given points". Please also make the same change in methods below.